r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+    geom_point()
+`r ''````
+````
+
+```{r dog-license-plot, echo=FALSE}
+ggplot(dog_licenses, aes(x = ..., y = ...)) +
+    geom_point()
+```
+
+What if we want to include the plot but not the code? Easy! Set the chunk option
+`echo` to `FALSE`:
+
+````markdown
+`r ''````{r dog-license-plot, echo=FALSE}
+ggplot(dog_licenses, aes(x = ..., y = ...)) +
+    geom_point()
+`r ''````
+````
+
+Or what if you don't want R to run the code chunk, but still show the code? Set
+the chunk option `eval` (for evaluate) to `FALSE`:
+
+````markdown
+`r ''````{r dog-license-plot, eval=FALSE}
+ggplot(dog_licenses, aes(x = ..., y = ...)) +
+    geom_point()
+`r ''````
+````
+
+Since we have a figure, we can change some width, height, and alignment options,
+as well as add a caption with the `fig.width`, `fig.height`, `fig.align`, and `fig.cap`
+(respectively):
+
+````markdown
+`r ''````{r dog-license-plot, fig.cap="Add your figure title here.", fig.height=8, fig.width=8}
+ggplot(dog_licenses, aes(x = ..., y = ...)) +
+    geom_point()
+`r ''````
+````
+
+```{r dog-license-plot-1, fig.cap="Add your figure title here.", fig.height=8, fig.width=8, echo=FALSE}
+ggplot(dog_licenses, aes(x = ..., y = ...)) +
+    geom_point()
+```
+
+What if we want to first run some code before running the plot, but don't want to 
+include the other code? Don't let it get evaluated so it outputs something. So,
+this will output the data frame and the plot:
+
+````markdown
+`r ''````{r dog-license-plot-after-2000, fig.cap="Add your figure title here.", fig.height=8, fig.width=8}
+dog_licenses_after_2000 <- dog_licenses %>% 
+    fitler(... >= 2000)
+dog_licenses_after_2000
+ggplot(dog_licenses_after_2000, aes(x = ..., y = ...)) +
+    geom_point()
+`r ''````
+````
+
+```{r dog-license-plot-after-2000, fig.cap="Add your figure title here.", fig.height=8, fig.width=8, echo=FALSE}
+dog_licenses_after_2000 <- dog_licenses %>% 
+    fitler(... >= 2000)
+dog_licenses_after_2000
+ggplot(dog_licenses_after_2000, aes(x = ..., y = ...)) +
+    geom_point()
+```
+
+While this will only output the plot. See how we don't include code that would 
+send anything to the console to be "printed"? Only things that get output get
+included in the R Markdown output document. Also notice how we renamed the chunk
+label? In R Markdown you can't have duplicate code chunk labels.
+
+````markdown
+`r ''````{r dog-license-plot-after-2000, fig.cap="Add your figure title here.", fig.height=8, fig.width=8}
+dog_licenses_after_2000 <- dog_licenses %>% 
+    fitler(... >= 2000)
+ggplot(dog_licenses_after_2000, aes(x = ..., y = ...)) +
+    geom_point()
+`r ''````
+````
+
+(ref:dog-plot-2000) Add your figure title here.
+
+```{r dog-license-plot-after-2000, fig.cap="(ref:dog-plot-2000)", fig.height=8, fig.width=8, echo=FALSE}
+dog_licenses_after_2000 <- dog_licenses %>% 
+    fitler(... >= 2000)
+ggplot(dog_licenses_after_2000, aes(x = ..., y = ...)) +
+    geom_point()
+```
+
+Neat!
```

Comment:
This is something I might say aloud, but it seems odd in print.

Reply:
Fair enough. Removed